src/Accordion/Accordion.tsx

```diff
@@ -1,95 +1,61 @@
     alternative?: boolean;
-    /** The index of an item that would be set to be expanded by default  */
-    defaultExpanded?: number;
-}
-
-const Accordion: React.FC<AccordionProps> = React.memo((props: AccordionProps) => {
-    const [active, setActive] = React.useState<number>(props.defaultExpanded);
-    const [idList, setIdList] = React.useState<Array<string>>([]);
-    const [id, setId] = React.useState<string>("");
-    const [className, setClassName] = React.useState<string>("");
-
-    React.useEffect(() => constructIds(), [props.list]);
-    React.useEffect(() => setActive(props.defaultExpanded), [props.defaultExpanded]);
+    /** An event handler triggered when an accordion toggle is clicked */
+    onToggle?: React.MouseEventHandler<HTMLButtonElement>;
+};
+
+export const Accordion: React.FC<AccordionProps> = React.memo(({ list, alternative, onToggle, ...props }: AccordionProps) => {
+    const [active, setActive] = React.useState<number>(Number(props.defaultValue));
+    const [id, setId] = React.useState<string>(props.id);
+    const [className, setClassName] = React.useState<string>("seb accordion");
+
+    React.useEffect(() => {
+        let defaultChecked: number;
+        defaultChecked = list?.findIndex((item: AccordionItemProps) => item.defaultChecked);
+        defaultChecked = React.Children.toArray(props.children).findIndex((Child: React.ReactElement<AccordionItemProps>) => Child.props?.defaultChecked);
+        setActive(defaultChecked >= 0 ? defaultChecked : props.defaultValue === null ? undefined : Number(props.defaultValue));
+    }, [props.defaultValue, list, props.children]);
     React.useEffect(() => setId(props.id || randomId("accordion-")), [props.id]);
-    React.useEffect(() => setClassName(classnames(["seb", "accordion", { alternative: props.alternative }, props.className])), [props.className, props.alternative]);
-
-    /** Constructs identifiers for accordion items to be used enable accessibility features */
-    const constructIds: VoidFunction = React.useCallback(() => {
-        const idListToSet: Array<string> = [];
-        props.list?.map(() => idListToSet.push(randomId("accordion-item-")));
-        setIdList(idListToSet);
-    }, [props.list, setIdList]);
+    React.useEffect(() => setClassName(classnames(["seb", "accordion", { alternative }, props.className])), [props.className, alternative]);
 
     /**
      * Handles accordion item click event
      * @param {React.MouseEvent<HTMLButtonElement>} e MouseEvent
      */
-    const onToggle: React.MouseEventHandler<HTMLButtonElement> = React.useCallback(
+    const onToggleInner: React.MouseEventHandler<HTMLButtonElement> = React.useCallback(
         (e: React.MouseEvent<HTMLButtonElement>) => {
-            setActive(active === Number(e.currentTarget.value) ? null : Number(e.currentTarget.value));
+            setActive(active === Number(e.currentTarget.dataset.indexNumber) ? null : Number(e.currentTarget.dataset.indexNumber));
```

Comment:
```suggestion
            setActive(active === Number(e.currentTarget.dataset.indexNumber) ? null : Number(e.currentTarget?.dataset?.indexNumber));
```

Reply:
This case will never happen because Accordion injects the "data-index-number" to any element. Even testing it is impossible since there is no case where it doesn't exist.